lib/spack/spack/architecture.py

```diff
@@ -587,6 +585,8 @@ def __call__(self):
 
 @contextlib.contextmanager
 def use_platform(new_platform):
+    import spack.compilers
+    import spack.config
```

Comment:
Does this mean that this refactor introduces new circular dependencies at module creation? Is this due to some function that moved from one module to another?

Reply:
Hm, weird, by now it's no long necessary... Maybe because I also deleted some imports elsewhere afterwards.